### PR TITLE
Implement focusGroup and unfocusGroup

### PR DESCRIPTION
**Pull request by alexweininger**
Branched off of https://github.com/microsoft/vscode-azureresourcegroups/pull/182, so I'll merge this after that is merged.

Implemented using [ExtensionContext.workspaceState](https://code.visualstudio.com/api/references/vscode-api#ExtensionContext)

I kept the auto reveal behavior, but I'm not totally sold on it.

https://user-images.githubusercontent.com/12476526/164131800-8a277a5d-ec08-40bc-813b-6bb5d8da4597.mov

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -165,6 +165,18 @@
                 "category": "Azure Resource Groups",
                 "icon": "$(refresh)"
             },
+            {
+                "command": "azureResourceGroups.focusGroup",
+                "title": "Focus Group",
```
Comment (bwateratmsft) [inline package.json:170]: Would like PM input on this too, but maybe "Focus this Group"? To me, "Focus Group" sounds like a noun before I realize it's meant to be an verb/noun combo.

File: src/extensionVariables.ts [hunk 2]
```diff
@@ -4,13 +4,21 @@
...
 import { AzExtTreeDataProvider, AzExtTreeItem, IAzExtOutputChannel } from "@microsoft/vscode-azext-utils";
-import { DiagnosticCollection, Disposable, ExtensionContext, TreeView } from "vscode";
+import { DiagnosticCollection, Disposable, Event, EventEmitter, ExtensionContext, TreeView } from "vscode";
 import { ActivityLogTreeItem } from "./activityLog/ActivityLogsTreeItem";
 import { AppResourceProvider } from "./api";
 import { TagFileSystem } from "./commands/tags/TagFileSystem";
 import { AzureAccountTreeItem } from "./tree/AzureAccountTreeItem";
 import { ExtensionActivationManager } from "./utils/ExtensionActivationManager";
 
+namespace extEmitters {
```
Comment (bwateratmsft) [inline src/extensionVariables.ts:14]: I like this idea!

File: src/tree/SubscriptionTreeItem.ts [hunk 3]
```diff
@@ -50,6 +50,13 @@ export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
         super(parent, subscription);
         this.resetCache();
         this.registerRefreshEvents('groupBy');
+
+        registerEvent('treeView.onDidChangeFocuedGroup', ext.events.onDidChangeFocusedGroup, async (context: IActionContext) => {
```
Comment (bwateratmsft) [inline src/tree/SubscriptionTreeItem.ts:54]: Nit
```suggestion
        registerEvent('treeView.onDidChangeFocusedGroup', ext.events.onDidChangeFocusedGroup, async (context: IActionContext) => {
```

**Review by bwateratmsft — APPROVED** [review]

File: src/extensionVariables.ts (hunk 2 above)
Comment (alexweininger) [reply src/extensionVariables.ts:14]: One thing to note here is that the order of the declaring the namespaces matters. I had them declared below `ext`, and TS didn't yell at me but they were always undefined at runtime.

**Review by bwateratmsft — APPROVED** [review]

**Pull request merged**